### ISSUE TITLE
TimerService.java implements equals but not hashCode

### ISSUE THREAD

**Issue by JonasDev** [Yakindu/statecharts#1002]
Overriding _equals_ requires implementing _hashCode_ as per _hashCode's_ contract. Otherwise a compiler warning is issued. Possible solution:
```
@Override
public int hashCode() {
	return Objects.hash(callback, eventID);
}
```

**Comment by andreasmuelder** [comment]
Thanks for the report. Hoewever, we should provide a classic hashCode implementation instead of using Objects to avoid dependencies to java 7 since our generated code is java 5 compatible.

**Comment by andreasmuelder** [comment]
Merged with PR #1016
